kernel/cClosure.ml

```diff
@@ -528,7 +544,16 @@ and to_constr_case (lfts,_ as ulfts) ci u pms p iv c ve env =
             Array.map f_ctx ve)
 
 and comp_subs (el,u) (s,u') =
-  Esubst.lift_subst (fun el c -> lazy (to_constr (el,u) c)) el s, u'
+  Esubst.lift_subst (fun el c ->
+      let elem = (el,u,c) in
+      match Cache.find elem Cache.cache with
+      | Some l -> l
```

Comment:
It may be useful to promote here

Reply:
Yeah, absolutely. My assumptions about the `lru` library all turned out to be wrong..